lib/saluki-app/src/memory.rs

```diff
@@ -255,3 +270,68 @@ pub async fn initialize_allocator_telemetry() -> Result<(), GenericError> {
 
     Ok(())
 }
+
+struct CgroupMemoryParser {}
+
+impl CgroupMemoryParser {
+    fn parse(self) -> Option<ByteSize> {
+        if let Ok(mut file) = fs::File::open("/proc/self/cgroup") {
+            let mut contents = String::new();
+            if file.read_to_string(&mut contents).is_ok() {
+                let parts: Vec<&str> = contents.trim().split("\n").collect();
+                // CgroupV2 has unified controllers.
+                if parts.len() == 1 {
+                    return self.parse_controller_v2(parts[0]);
+                }
+                for line in contents.lines() {
```

Comment:
You already have `parts` where you've split the file into separate lines... I think you can just that here instead of `contents.lines()`.